extensions/cocostudio/loader/load.js

```diff
@@ -193,6 +193,36 @@ ccs.load = function(file, path){
     return object;
 };
 
+/**
+ * Analysis of studio JSON file and layout ui widgets by visible size.
+ * The incoming file name, parse out the corresponding object
+ * Temporary support file list:
+ *   ui 1.*
+ *   node 1.* - 2.*
+ *   action 1.* - 2.*
+ *   scene 0.* - 1.*
+ * @param {String} file
+ * @param {String} [path=] Resource path
+ * @returns {{node: cc.Node, action: cc.Action}}
+ */
+ccs.loadWithVisibleSize = function(file, path){
+    var object = {
+        node: null,
+        action: null
+    };
+
+    object.node = ccs._load(file, null, path);
+    var size = cc.director.getVisibleSize();
+    if(object.node && size){
+        object.node.setContentSize(size.width, size.height);
+        ccui.helper.doLayout(object.node);
+    }
+    object.action = ccs._load(file, "action", path);
+    if(object.action && object.action.tag === -1 && object.node)
+        object.action.tag = object.node.tag;
+    return object;
+};
+
```

Comment:
I think it could be written:
  ccs.loadWithVisibleSize = function(file, path){
    var obj = ccs.load(file, path);
    var node = obj.node, size = cc.director.getVisibleSize();
    if(node && size){
      xxx
    }
    return obj;
  }